test/index.test.js

```diff
@@ -72,9 +109,21 @@ describe('index', () => {
         Executor = require('../index');
 
         executor = new Executor({
-            host: 'jenkins',
-            username: 'admin',
-            password: 'fakepassword'
+            ecosystem,
+            jenkins: {
+                host: 'jenkins',
+                username: 'admin',
+                password: 'fakepassword',
+                nodeLabel
+            },
+            buildScript,
+            cleanupScript,
+            cleanupWatchInterval
+        });
+        compiledJobXml = _.template(TEST_XML)({
```

Comment:
This looks like you are generating the test result and the expected result with the same code. Why not make the TEST_XML have the correct expected data in it to begin with, or a different xml file with the expected data?